### PR TITLE
build: Vendor CocoaAsyncSocket, CocoaHTTPServer and RoutingHTTPServer

### PR DESCRIPTION
**Pull request by qmfrederik**
CocoaHTTPServer and RoutingHTTPServer are not being actively maintained and have been forked by the Appium team.

Instead of maintaining code in three separate repositories, consolidate the CocoaHTTPServer and RoutingHTTPServer in the WebDriverAgent repository.

CocoaHTTPServer contains a source-level copy of CocoaAsyncSocket, so this code has been included, too.

This PR:
- Copies the CocoaAsyncCode, CocoaHTTPServer and RoutingHTTPServer to the `WebDriverAgentLib/Vendor` folder
- Adds the BSD-style licenses for these projects
- Disables certain compiler warnings the sources which have been copied
- Removes unused code (File responses, multi-part forms, HTTP authentication, HTTPS)
- Updates the README with an attribution notice

This PR contains individual commits to keep the history easy to digest. Feel free to squash these commits if you want.

This PR should compile and all tests should pass. There's some work that can still be done, preferably in follow-up PRs:
- Upgrade to the latest version of CocoaAsyncSockets
- Take a dependency on the upstream CocoaAsyncSockets project (which is still being maintained) instead of copying the sources
- Fix compiler warnings instead of silencing them

### CODE REVIEW

**Comment by mykola-mokhnach** [comment]
Would it be possible to autoformat the lib files, so they follow the common WDA indentation style (2 spaces)?

**Comment by qmfrederik** [comment]
Sure. I didn't reformat CocoaAsyncSocket because they come from a maintained upstream (see PR description).

**Comment by mykola-mokhnach** [comment]
> Sure. I didn't reformat CocoaAsyncSocket because they come from a maintained upstream (see PR description).

What if we include CocoaAsyncSocket to the carthage dependencies then? If you say it is maintained properly then it makes sense to use that snapshot rather than copy it here. Or RoutingHTTPServer cannot work with the recent version of it?

**Comment by qmfrederik** [comment]
> What if we include CocoaAsyncSocket to the carthage dependencies then?

Yes, I think that's the proper way to fix it. I've tried upgrading it to the latest version (by copying the source), and it worked just fine on iOS. There were some build errors on tvOS but nothing fundamental, I think.

That said, this PR in its current state is "pure refactoring" - almost no source code changes. Upgrading CocoaAsyncSockets will include a significant amount of code change, and that's why it's perhaps better to do it on a separate PR?

What do you think?

**Comment by mykola-mokhnach** [comment]
yep, that makes sense. Lets do it in separate PR.

**Review by mykola-mokhnach — APPROVED** [review]

File: Cartfile [hunk 1]
```diff
@@ -1,5 +1,2 @@
-# Used for HTTP routing
-github "appium/RoutingHTTPServer"
```
Comment (KazuCocoa) [inline Cartfile:2]: 👍

**Review by KazuCocoa — APPROVED** [review]

**Pull request merged**